apps/veyfi/contexts/useVotingEscrow.tsx

```diff
@@ -57,87 +53,102 @@ const defaultProps: TVotingEscrowContext = {
-		const veYFIContract = new Contract(VEYFI_ADDRESS, VEYFI_ABI);
-		const [token, name, symbol, decimals, supply, rewardPool] = await ethcallProvider.tryAll([
-			veYFIContract.token(),
-			veYFIContract.name(),
-			veYFIContract.symbol(),
-			veYFIContract.decimals(),
-			veYFIContract.supply(),
-			veYFIContract.reward_pool()
-		]) as [TAddress, string, string, number, BigNumber, TAddress];
-
+	const {address, isActive} = useWeb3();
+
+	/* 🔵 - Yearn Finance **********************************************************
+	** Retrieving the basic information of the veYFI contract. They are not linked
+	** to the user's address, so they are not affected by the `isActive` flag.
+	******************************************************************************/
+	const baseVeYFIContract = {address: toWagmiAddress(VEYFI_ADDRESS), abi: VEYFI_ABI};
+	const {data: votingEscrowData, status: votingEscrowStatus, refetch: refreshVotingEscrow} = useContractReads({
+		contracts: [
+			{...baseVeYFIContract, functionName: 'token'},
+			{...baseVeYFIContract, functionName: 'name'},
+			{...baseVeYFIContract, functionName: 'symbol'},
+			{...baseVeYFIContract, functionName: 'decimals'},
+			{...baseVeYFIContract, functionName: 'supply'},
+			{...baseVeYFIContract, functionName: 'reward_pool'}
+		]
+	});
+	const votingEscrow = useMemo((): TVotingEscrow | undefined => {
+		if (!votingEscrowData || votingEscrowStatus !== 'success') {
+			return undefined;
+		}
+		const [token, name, symbol, decimals, supply, rewardPool] = votingEscrowData;
 		return ({
 			address: VEYFI_ADDRESS,
-			token,
-			name,
-			symbol,
-			decimals,
-			supply,
-			rewardPool
+			token: toAddress(decodeAsString(token)),
+			name: decodeAsString(name),
+			symbol: decodeAsString(symbol),
+			decimals: decodeAsNumber(decimals),
+			supply: decodeAsBigInt(supply),
+			rewardPool: toAddress(decodeAsString(rewardPool))
 		});
-	}, [provider]);
-	const {data: votingEscrow, mutate: refreshVotingEscrow, isLoading: isLoadingVotingEscrow} = useSWR('asset', assetFetcher, {shouldRetryOnError: false});
-
-	const positionsFetcher = useCallback(async (): Promise<TVotingEscrowPosition> => {
-		if (!isActive || !address) {
-			return {};
+	}, [votingEscrowData, votingEscrowStatus]);
+
+
+	/* 🔵 - Yearn Finance **********************************************************
+	** Retrieving the user's positions in the veYFI contract. They are linked to the
+	** user's address, so they are affected by the `isActive` flag.
+	******************************************************************************/
+	const baseVeYFIPositionContract = {address: toWagmiAddress(VEYFI_POSITION_HELPER_ADDRESS), abi: VEYFI_POSITION_HELPER_ABI};
+	const {data: positionData, status: positionStatus, refetch: refreshPosition} = useContractRead({
+		...baseVeYFIPositionContract,
+		functionName: 'getPositionDetails',
+		args: [toWagmiAddress(address)],
+		enabled: isActive && address !== undefined && !isZeroAddress(address)
```

Comment:
Maybe making this comparison a bit more broad `!!address`, unsure if `isZeroAddress` will take a `null` for example